vignettes/decorate-modules-output.Rmd

```diff
@@ -0,0 +1,271 @@
+    )
+  },
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive(stop("This is error"))
+    })
+  }
+)
+    ```
+
+## Example Module
+
+To include decorators in a `teal` module, pass them as arguments (`ui_args` and `server_args`) to the module’s `ui` and 
+`server` components, where they will be used by `ui/srv_teal_transform_module`:
+
+
+```{r}
+tm_decorated_plot <- function(label = "module", transforms = list(), decorators = teal_transform_module()) {
+  module(
+    label = label,
+    ui = function(id, decorators) {
+      ns <- NS(id)
+      div(
+        selectInput(ns("dataname"), label = "select dataname", choices = NULL),
+        selectInput(ns("x"), label = "select x", choices = NULL),
+        selectInput(ns("y"), label = "select y", choices = NULL),
+        ui_teal_transform_module(ns("decorate"), transforms = decorators),
+        plotOutput(ns("plot")),
+        verbatimTextOutput(ns("text"))
+      )
+    },
+    server = function(id, data, decorators) {
+      moduleServer(id, function(input, output, session) {
+        observeEvent(data(), {
+          updateSelectInput(inputId = "dataname", choices = teal.data::datanames(data()))
+        })
+
+        observeEvent(input$dataname, {
+          updateSelectInput(inputId = "x", choices = colnames(data()[[input$dataname]]))
+          updateSelectInput(inputId = "y", label = "select y", choices = colnames(data()[[input$dataname]]))
+        })
+
+        q1 <- reactive({
+          req(input$dataname, input$x, input$y)
+          data() |>
+            within(
+              {
+                plot <- ggplot2::ggplot(dataname, ggplot2::aes(x = x, y = y)) +
+                  ggplot2::geom_point()
+              },
+              dataname = as.name(input$dataname),
+              x = as.name(input$x),
+              y = as.name(input$y)
+            )
+        })
+
+        q2 <- srv_teal_transform_module("decorate", data = q1, transforms = decorators)
+
+        plot_r <- reactive({
+          req(q2())
+          q2()[["plot"]]
+        })
+
+        output$plot <- renderPlot(plot_r())
+        output$text <- renderText({
+          teal.code::get_code(q2())
+        })
+      })
+    },
+    ui_args = list(decorators = decorators),
+    server_args = list(decorators = decorators)
+  )
+}
+```
+
+## Example app
+
+```{r}
+library(ggplot2)
+app <- init(
+  data = teal_data(iris = iris, mtcars = mtcars),
+  modules = modules(
+    tm_decorated_plot("identity"),
+    tm_decorated_plot("no-ui", decorator = static_decorator),
+    tm_decorated_plot("lang", decorator = static_decorator_lang),
+    tm_decorated_plot("interactive", decorator = interactive_decorator),
+    tm_decorated_plot("interactive-from lang", decorator = interactive_decorator_lang),
+    tm_decorated_plot("from-fun", decorator = gg_xlab_decorator("plot")),
+    tm_decorated_plot("failing", decorator = failing_decorator)
```

Comment:
Those decorators are not created in this vignette. Are they distributed with `teal`? Should we include them here